server/src/main/java/io/crate/planner/operators/LogicalPlanner.java

```diff
@@ -417,6 +417,9 @@ public LogicalPlan visitAliasedAnalyzedRelation(AliasedAnalyzedRelation relation
         public LogicalPlan visitView(AnalyzedView view, List<Symbol> outputs) {
             var child = view.relation();
             var source = child.accept(this, child.outputs());
+            if (source instanceof ForeignCollect foreignCollect) {
```

Comment:
This doesn't compose well. The `visitView` handling must stay unaware of concrete child implementations.

How about instead using the `sessionUser` mechanism of the coordinatorSessionSettings. E.g. this here could look like:

```java
CoordinatorSessionSettings sessionSettings = coordinatorTxnCtx.sessionSettings();
// TODO: resolve owner to role in analyzer?
Role sessionUser = sessionSettings.sessionUser();
sessionSettings.setSessionUser(view.owner());
var source = child.accept(this, child.outputs());
sessionSettings.setSessionUser(sessionUser);
```

Then the `visitForeignTable` can query the session user, and pass it to the `ForeignCollect` constructor.

Reply:
Thank you, regarding

> // TODO: resolve owner to role in analyzer?

It seems that we have been analyzing users in later steps, maybe this could be done during feature freeze?